EnvironmentSimulator/Modules/RoadManager/RoadManager.cpp

```diff
@@ -8274,15 +8274,24 @@ bool Position::Delta(Position* pos_b, PositionDiff &diff, bool bothDirections, d
 		int laneIdB = pos_b->GetLaneId();
 		double tB = pos_b->GetT();
 
-		// If start and end roads are oppotite directed, inverse one side for delta calculations
-		if (path->visited_.size() > 0 &&
-			((path->visited_[0]->link->GetType() == LinkType::SUCCESSOR &&
-				path->visited_.back()->link->GetContactPointType() == ContactPointType::CONTACT_POINT_END) ||
-			 (path->visited_[0]->link->GetType() == LinkType::PREDECESSOR &&
-				path->visited_.back()->link->GetContactPointType() == ContactPointType::CONTACT_POINT_START)))
-		{
-			laneIdB = -laneIdB;
-			tB = -tB;
+		if (path->visited_.size() > 0) {
+			RoadPath::PathNode* lastNode = path->visited_.back();
+			RoadPath::PathNode* firstNode = lastNode;
+			while (firstNode->previous) {
```

Comment:
I understand this loop. It seems like a flaw that visited[0] is not the actual first road in the "path".
Still it would be nice to skip the loop. I'll check later if there is another solution (to store FIRST road).

Reply:
Yes, I like the idea to store the already calculated start direction. And then re-use it the Delta method which could be used frequently, hence we want to minimize unnecessary work.

I took the liberty of committing an attempt to achieve this on your branch. If it builds, please test and report how it works.